src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -190,15 +190,27 @@ abstract class ComputeAdviceTask @Inject constructor(
 
       output.bufferWriteJson(projectAdvice)
       // These must be transformed so that the Coordinates are Strings for serialization
-      dependencyUsagesOut.bufferWriteJsonMap(dependencyUsages.toStringCoordinates())
-      annotationProcessorUsagesOut.bufferWriteJsonMap(annotationProcessorUsages.toStringCoordinates())
+      dependencyUsagesOut.bufferWriteJsonMap(dependencyUsages.toStringCoordinates(buildPath))
+      annotationProcessorUsagesOut.bufferWriteJsonMap(annotationProcessorUsages.toStringCoordinates(buildPath))
       // TODO consider centralizing this logic in a separate PR
       bundleTraces.bufferWriteJsonSet(dependencyAdviceBuilder.bundledTraces)
     }
 
-    private fun Map<Coordinates, Set<Usage>>.toStringCoordinates(): Map<String, Set<Usage>> = map { (key, value) ->
-      key.gav() to value
-    }.toMap()
+    private fun Map<Coordinates, Set<Usage>>.toStringCoordinates(buildPath: String): Map<String, Set<Usage>> =
+      map { (key, value) ->
+        toCoordinatesKey(key, buildPath) to value
+      }.toMap()
+
+    private fun toCoordinatesKey(coordinates: Coordinates, buildPath: String) =
+      coordinates.gav() + when {
+        // If the coordinates point at a project in the current build, there are two notations to address it:
+        // 1) 'group:name' coordinates, which is the main representation we aim to use everywhere (see IncludedBuildCoordinates)
```

Comment:
Do we aim to use it everywhere?

Reply:
I removed that part of the comment. I am also not sure what I meant to say with that. Thanks.